qcelemental/models/molecule.py

```diff
@@ -74,44 +74,172 @@ class Config(ProtoModel.Config):
 
 
 class Molecule(ProtoModel):
+    """
+    The fundamental representation of a Molecule inside the QCArchive Ecosystem. This model contains
+    data for atomic elements, coordinates, connectivity, charges, fragmentation, etc. The model also
+    supports orientation and measurement utilities.
+
+    All Molecule objects are oriented via inertia tensor with the center-of-mass at (0,0,0) unless
+    explicitly blocked.
+    """
 
-    schema_name: constr(strip_whitespace=True, regex=qcschema_molecule_default) = qcschema_molecule_default
-    schema_version: int = 2
-    validated: bool = False
+    schema_name: constr(strip_whitespace=True, regex=qcschema_molecule_default) = Schema(
+        qcschema_molecule_default,
+        description=(f"The QCSchema specification this model conforms to. Explicitly fixed as "
+                     f"{qcschema_molecule_default}.")
+    )
+    schema_version: int = Schema(
+        2,
+        description="The version number of ``schema_name`` that this Molecule model conforms to."
+    )
+    validated: bool = Schema(
+        False,
+        description="A boolean indicator that the input Molecule data has been previously checked against"
+                    "schema for speed purposes. This should be False in most cases. A ``True`` setting "
+                    "should only ever be set by the constructor for this class itself or other trusted sources such as "
+                    "a Fractal Server or previously serialized Molecules."
+    )
 
     # Required data
-    symbols: Array[str]
-    geometry: Array[float]
+    symbols: Array[str] = Schema(
+        ...,
+        # LNN Note: "Ordered" might not be the least ambiguous term
+        # (e.g. "ordered" as in linked ordering or "ordered" as in alphabetical?)
+        description="An ordered 1-D array-like object of atomic elemental symbols of shape (N). Index of this list "
+                    "sets atomic order for all other per-atom setting like ``real`` and the first dimension of "
+                    "``geometry``. Ghost/virtual atoms "
+                    "must have an entry in this array-like and are indicated by the matching index in ``real``"
+                    "field."
+    )
+    geometry: Array[float] = Schema(
+        ...,
+        description="An ordered 2-D array-like object of shape (N,3) for atomic coordinates. Index of the "
+                    "first dimension matches the index of all other per-atom settings like ``symbols`` and ``real``."
+    )
 
     # Molecule data
-    name: str = ""
-    identifiers: Optional[Identifiers] = None
-    comment: Optional[str] = None
-    molecular_charge: float = 0.0
-    molecular_multiplicity: int = 1
+    name: str = Schema(
+        "",
```

Comment:
optional, not "", I think

Reply:
Yup.

Reply:
This has been a non-optional string before now which just defaulted to the empty string.. I made it optional, we'll see if it breaks anything